packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -0,0 +1,300 @@
+      contentPosition.vertical
+    );
+    const contentHorizontalOffset = getOffsetLeft(
+      contentRect,
+      contentPosition.horizontal
+    );
+
+    top -= contentVerticalOffset;
+    left -= contentHorizontalOffset;
+
+    // Move menu back into view if out of screen
+    const { marginThreshold = 0 } = this.props;
+    const viewContainer = getContainer(anchorEl);
+
+    const heightMax = viewContainer.innerHeight - marginThreshold;
+    const widthMax = viewContainer.innerWidth - marginThreshold;
+
+    // Check Vertical Constraints
+    if (top + contentRect.height > heightMax) {
+      top -= top + contentRect.height - heightMax;
+    } else if (top < marginThreshold) {
+      top = marginThreshold;
+    }
+
+    // Check Horizontal Constraints
+    if (left + contentRect.width > widthMax) {
+      left -= left + contentRect.width - widthMax;
+    } else if (left < marginThreshold) {
+      left = marginThreshold;
+    }
+
+    return {
+      top: `${top}px`,
+      left: `${left}px`,
+      position: "fixed"
+    };
+  };
+
+  setContentElRef = el => {
+    this.setState({
+      contentRef: el
+    });
+  };
+
+  isDecendant(parent, child) {
+    const node = child.parentNode;
+    if (node == null) {
+      return false;
+    } else if (node === parent) {
+      return true;
+    }
+    return this.isDecendant(parent, node);
+  }
+
+  closeEvent = event => {
+    const { isOpen } = this.props;
+    const { contentRef } = this.state;
+    if (!contentRef || !isOpen) return;
+    if (event.type === "mousedown") {
+      // Check if mouse click happened inside popover
+      const target = event.target;
+      if (contentRef === target || this.isDecendant(contentRef, event.target)) {
+        return;
+      }
+      event.preventDefault();
+      this.handleClose();
+    } else if (event.type === "keydown" && event.key === "Escape") {
+      event.preventDefault();
+      this.handleClose();
+    }
+  };
+
+  handleClose = () => {
+    const { onClose } = this.props;
+    if (onClose) onClose();
+  };
+
+  // Find the outer most scrollable content and remove the scrollbar
+  lockParentScroll() {
+    const anchorEl = this.getAnchorEl();
+    const parent = anchorEl.parentElement;
+    const container = getDocument(anchorEl);
+
+    this.state.scrollContainer =
+      parent.nodeName === "HTML" &&
+      window.getComputedStyle(parent)["overflow-y"] === "scroll"
+        ? parent
+        : container.body;
+
+    this.state.scrollContainerStyle = this.state.scrollContainer.style.overflow;
+    this.state.scrollContainer.style.overflow = "hidden";
+  }
+
+  resetParentScroll() {
+    this.state.scrollContainer.style.overflow = this.state.scrollContainerStyle;
+  }
+
+  updatePosition = () => {
+    if (!this.props.anchorEl) return undefined;
+    return this.forceUpdate();
+  };
+
+  static contextType = ThemeContext;
+
+  render() {
+    const { classes, children, type = "span", anchorEl, isOpen } = this.props;
```

Comment:
Instead of doing `type = 'span'` here, move this to be a `defaultProp`. This way we don't have to remember to destructure this default value as a `span` if we use it anywhere else in the future.

Reply:
👍